### ISSUE TITLE
Error when sample_rate < 1

### ISSUE THREAD

**Issue by TranThanh96** [HaiyuWu/SOTA-Face-Recognition-Train-and-Test#2]
I am trying to train with wf42m, I set sample_rate=0.3 and train with uniface, I think error at:
```
loss = one_hot * p_loss + (~one_hot) * n_loss
```
can you check it ?

**Comment by TranThanh96** [comment]
I think you missing this one: https://github.com/CVI-SZU/UniFace/blob/main/head.py#L116
```
one_hot = torch.index_select(one_hot, 1, partial_index)
```

**Comment by HaiyuWu** [comment]
one_hot.scatter_(1, label.view(-1, 1).long(), 1)
        one_hot = torch.index_select(one_hot, 1, partial_index)
is equal to
        one_hot.scatter_(1, labels[index].view(-1, 1).long(), 1)
